netcam_rtsp.c

```diff
@@ -456,6 +452,23 @@ static int netcam_rtsp_open_context(netcam_context_ptr netcam){
         return -1;
     }
 
+    /*
+     * Documentation does not indicate thread safety on these
+     * init functions so we lock them just in case.
+     */
+    if (netcam->rtsp->status == RTSP_RECONNECTING){
+        pthread_mutex_lock(&global_lock);
+            avformat_network_deinit();
```

Comment:
The warning message about a lack of calls to avformat_network_init (at least in the current ffmpeg) could only happen if:

1) avformat_network_deinit has been explicitly called
or:
2) we've corrupted the memory where it's ff_network_inited_globally flag is stored

(see https://github.com/FFmpeg/FFmpeg/blob/267da70ea8c36caaa645a3c4f1c5f0ca8bae156a/libavformat/utils.c#L4662 and https://github.com/FFmpeg/FFmpeg/search?utf8=✓&q=ff_network_inited_globally )

It seems 1) isn't the case, at least from reviewing the current code. So 2) seems likely. I don't see any other possible explanation.

If it's 2) I'd like to get to the bottom of the root cause rather than trying to patch the symptoms.